app/controllers/publishers_controller.rb

```diff
@@ -76,19 +73,26 @@ def verify
     ).perform
     current_publisher.reload
     if current_publisher.verified?
-      flash.notice = I18n.t("publishers.verify_success")
       render(:verification_done)
     else
-      i18n_name = "publishers.verify_failed_#{current_publisher.verification_method}"
-      flash.now[:alert] = I18n.t(i18n_name)
-      render(publisher_verification_action)
+      render(:verification_failed)
     end
   rescue PublisherVerifier::VerificationIdMismatch
-    flash.now[:alert] = I18n.t("activerecord.errors.models.publisher.attributes.brave_publisher_id.taken")
-    render(publisher_verification_action)
+    @failure_reason = I18n.t("activerecord.errors.models.publisher.attributes.brave_publisher_id.taken")
+    render(:verification_failed)
   rescue Faraday::Error
-    flash.now[:alert] = I18n.t("shared.api_error")
-    render(publisher_verification_action)
+    @try_again = true
+    @failure_reason = I18n.t("shared.api_error")
+    render(:verification_failed)
+  end
+
+  # Verification is still in progress
+  def verification_failed
```

Comment:
i noticed you render this action's view above.
can the user ever visit this action? otherwise you could remove this action, and still use the view above.